src/parser/wsdl.js

```diff
@@ -104,8 +106,23 @@ class WSDL {
         self.xmlnsInEnvelope = self._xmlnsMap();
         callback(err, self);
       });
+    }
 
+    if (self.syncLoad) {
+      loadUpSchemas();
+    } else {
+      process.nextTick(loadUpSchemas);
+    }
+  }
+
+  loadSync() {
+    this.syncLoad = true;
+    var result;
+    this.load(function (err, wsdl) {
+        result = wsdl;
```

Comment:
Should we set `syncLoad` back to false in here and also return result since `this.load` is async?

Reply:
If I pulled the syncload in I'd need to pass in `var self=this`. and then `self.syncLoad=false` But it won't make a difference in the operation of the code as this load is being completed before the next lines are exectued.

The return needs to be outside to return to the calling function the result which has been set in the inner call.

I'll add a comment though to explain what is happening as its not intuitive!

Reply:
yeah that's true you'd need to do `var self = this`. I guess my main concern is say if `this.load` takes a while to execute, then the next lines will be executed before result is populated. If that is not a conern, then I'm okay to have it as is.

**EDIT**: I see the comment now 👍.

Reply:
Yes, if any code is introduced which causes this to go async for any reasonthen the tests we have added will fail as they will return result as undefined here. We've been using the code successfully in our mapper using this call and its passing in big schemas and its working well :-)